backend/src/permissions/permissions.service.ts

```diff
@@ -38,6 +40,43 @@ export class PermissionsService {
     private noteConfig: NoteConfig,
     private eventEmitter: EventEmitter2<NoteEventMap>,
   ) {}
+  /**
+   * Checks if the given {@link User} is has the in {@link desiredPermission} specified permission on {@link Note}.
+   *
+   * @async
+   * @param {Permission} desiredPermission - permission level to check for
+   * @param {User} user - The user whose permission should be checked. Value is null if guest access should be checked
+   * @param {Note} note - The note for which the permission should be checked
+   * @return if the user has the specified permission on the note
+   */
+  public async checkPermissionOnNote(
+    desiredPermission: Exclude<Permission, Permission.CREATE>,
+    user: User | null,
+    note: Note,
+  ): Promise<boolean> {
+    switch (desiredPermission) {
+      case Permission.READ:
+        return await this.mayRead(user, note);
+      case Permission.WRITE:
+        return await this.mayWrite(user, note);
+      case Permission.OWNER:
+        return await this.isOwner(user, note);
+    }
+    return false;
```

Comment:
You could test this line by introducing a new test case that calls this method with `            'UNKNOWN' as unknown as Permission.READ` as permission. Yes, I know this is an ugly type hack, but it works as intended.

Reply:
`UNKNOWN as keyof typeof Permission`?

Reply:
or you scrap the "return false" line.